methods/inputs/filter_gedi_data.py

```diff
@@ -9,85 +9,125 @@
+    with open(granule_list_file, "r") as file:
+        # read each line as a granule name and return the list
+        granule_files = file.read().splitlines()
+    return granule_files
+
+# main function to filter GEDI data based on the buffer and granule list
 def filter_gedi_data(
-    boundary_file: str,
     gedi_directory_path: str,
+    boundary_file: str,
+    granule_list_file: str,
     output_filename: str
 ) -> None:
 
+    # load the boundary file (e.g., shapefile or geojson)
     boundary = gpd.read_file(boundary_file)
 
-    # These are the columns we care about for calculating AGB values, the rest we'll drop
-    # to save memory/disk space
-    interesting_columns = {'sensitivity', 'geometry', 'absolute_time', 'l4_quality_flag', 'degraded_flag',
-        'beam_type', 'agbd'}
+    # specify the columns that are important for AGB calculations
+    interesting_columns = {'sensitivity', 'geometry', 'absolute_time', 'l4_quality_flag', 'degrade_flag',
+                           'beam_type', 'agbd'}
+
+    granules = []  # list to collect data from all granules
+    # load granule names from the provided text file
+    granule_files = load_granules_from_file(granule_list_file)
+
+    # loop through each granule name from the file
+    for granule_name in granule_files:
+        granule_path = os.path.join(gedi_directory_path, granule_name)  # create the full path to the granule
+        if os.path.exists(granule_path):  # check if the granule file exists
+            df = parse_file(granule_path)  # parse the granule file into a DataFrame
 
-    granules = []
-    for path in glob.glob("*.h5", root_dir=gedi_directory_path):
-        df = parse_file(os.path.join(gedi_directory_path, path))
+            # drop any columns not in the interesting_columns set
+            df.drop(list(set(df.columns) - interesting_columns), axis=1)
 
-        # Drop data we don't care about
-        # The date limits here come from discussion with Miranda Lam:
-        # "we currently use the JRC Annual Change 2020 for land cover class and GEDI shots from 2020/1/1 to 2021/1/1"
-        df.drop(list(set(df.columns) - interesting_columns), axis=1)
-        df.drop(df.index[df.beam_type != "full"], inplace=True)
-        df.drop(df.index[df.degrade_flag != 0], inplace=True)
-        df.drop(df.index[df.l4_quality_flag != 1], inplace=True)
-        df.drop(df.index[df.absolute_time < datetime(2020, 1, 1, tzinfo=timezone.utc)], inplace=True)
-        df.drop(df.index[df.absolute_time >= datetime(2021, 1, 1, tzinfo=timezone.utc)], inplace=True)
-        df.drop(df.index[~df.within(boundary.geometry.values[0])], inplace=True)
+            # filter rows based on beam type, quality flag, degradation flag, and date range
+            df.drop(df.index[df.beam_type != "full"], inplace=True)
+            df.drop(df.index[df.degrade_flag != 0], inplace=True)
+            df.drop(df.index[df.l4_quality_flag != 1], inplace=True)
```

Comment:
Why are the time value checks dropped? Doesn't this mean we'll now get shots that are from a different year to the JRC tiles we're comparing with?

If this is intentional does this need a methodology document update to go with it?

Reply:
Ah I see also the notes don't align with the code here anymore, this was a deliberate move as the time values are now specified in the `locate_gedi_data.py` script, meaning that only GEDI granules within the time frame specified in the new call for `locate_gedi_data.py` are loaded.

Reply:
Engaging pedant mode :) 

We download granules from NASA, which are a swathe of GEDI shots from a pass of the ISS. The filter in locate_gedi_data.py I *think* will return granules that contain data in that range, but I don't think it guarantees exclusivity - a sweep might start in one year and end in another. And so I think we still need the time checks here.

The likelihood of this happening is slim obviously, but not zero I think.